crates/cranelift/Cargo.toml

```diff
@@ -31,3 +32,4 @@ thiserror = { workspace = true }
 all-arch = ["cranelift-codegen/all-arch"]
 component-model = ["wasmtime-environ/component-model"]
 incremental-cache = ["cranelift-codegen/incremental-cache"]
+versioned-exports = ["wasmtime-versioned-export-macros/enabled", "wasmtime-environ/versioned-exports"]
```

Comment:
Ideally this could always be enabled and handled internally, so could the goal for the end state of this PR be to remove this as an option?

If this is only an issue on s390x I think that's a targeted problem which can be fixed in isolation in the build script for the outlined assembly files.

Reply:
I think it could makes sense not to use a feature for this. That being said, in discussion with @saulecabrera he mentioned that we likely want to support [`ModuleVersionStrategy::Custom`](https://docs.rs/wasmtime/latest/wasmtime/enum.ModuleVersionStrategy.html#) and also `ModuleVersionStrategy::None`, and having this enabled by default would, from my understanding, break those two settings. Should it be opt-in, such as enabled with a `WASMTIME_VERSIONED_EXPORTS` environment variable to use the version-based suffix and `WASMTIME_EXPORT_SUFFIX` for a custom suffix?

Reply:
Oh? In my mind this setting is purely about how we need some unmangled symbol names for things written in assembly and we need to be able to have "mangled" symbols per-wasmtime-version. That I would not expect to be related to the `ModuleVersionStrategy` which is only about what compiled artifacts look like, and as long as the right artifact is paired with the right wasmtime everything should work there as expected.

Reply:
My line of thought here was mostly around verifying if the usage of `ModuleVersionStrategy` would be in conflict with versioning the symbols with the crate's version. But I don't necessarily have a particular use-case that demonstrates such conflict, I mostly wanted to make sure that we're all aligned that this is not a concern, which I think @alexcrichton has clarified in the previous comment.

Reply:
Got it, sounds good. Thanks!

Reply:
Ah yeah I think we're all good then because the names of symbols in Wasmtime doesn't actually end up relating to compiled artifacts at all, so these end up being orthogonal knobs.